Zend/tests/gh10377_3.phpt

```diff
@@ -0,0 +1,10 @@
+--TEST--
+GH-10377 (Unable to have an anonymous readonly class) - usage variation: final modifier
+--FILE--
+<?php
+
+$x = new final class {};
+
+?>
+--EXPECTF--
+Fatal error: Cannot use the final modifier on an anonymous class in %s on line %d
```

Comment:
BTW it has just come to my mind that anonymous classes could be made final implicitly. Of course this only makes sense if the benefit is worth (better optimizations).

Reply:
I did a PR where this optimisation is done, but not by setting the flag but by checking the case explicitly: https://github.com/php/php-src/pull/10467

Reply:
Oh I see! I even had a look at it 🤦